### PR TITLE
chore: Clean up `src/utilities` exports

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝
- Removes default exports in favor of named exports ✏️
- Removes some unused helper functions 🗑️
- No logic changes or refactors, only find and replace of imports and exports 🚫

### Verification steps 
- Verify all Github Actions checks pass ✅
- Verify e2es pass 🤞🏼
- Open Cloud Manager and perform an overall check of the application for defects 👁️

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/App.tsx [hunk 1]
```diff
@@ -292,18 +291,3 @@ const BaseApp = withDocumentTitleProvider(
...
-      [0, 'reason'],
-      eachError
-    );
-    return typeof cleanedError !== 'string'
-      ? false
-      : cleanedError.toLowerCase().includes('oauth');
-  });
-};
-
-export const isOSMac = navigator.userAgent.includes('Mac');
```
Comment (bnussman-akamai) [inline packages/manager/src/App.tsx:309]: Moved this to `src/utilities/userAgent.ts`

File: packages/manager/src/utilities/arrayToList.ts [hunk 2]
```diff
@@ -1,4 +1,7 @@
-export default (input: string[], separator: string = ','): string => {
```
Comment (bnussman-akamai) [inline packages/manager/src/utilities/arrayToList.ts:1]: This is the case against default exports and why named exports provide better DX. We are exporting an unnamed function which makes our intelliesense unable to auto import it

File: packages/manager/src/utilities/isPropValid.ts
Comment (bnussman-akamai) [inline packages/manager/src/utilities/isPropValid.ts:1]: I moved some utilities out of their own folder in cases where the folder is unnecessary

In my opinion, this is bad 👎🏼

```bash
packages/manager/src/utilities/isPropValid
├── index.ts
└── isPropValid.ts
```

and this is good 👍🏼

```bash
packages/manager/src/utilities/pricing
├── backups.test.tsx
├── backups.ts
├── constants.ts
├── dynamicPricing.test.ts
├── dynamicPricing.ts
├── dynamicVolumePrice.test.ts
├── dynamicVolumePrice.ts
├── kubernetes.test.tsx
├── kubernetes.ts
├── linodes.test.ts
└── linodes.ts
```

**Review by jdamore-linode — APPROVED** [review]
Thanks @bnussman-akamai! Confirmed that the E2E tests look good (you can disregard the CI failures; DBaaS failures should be fixed in `develop`, and the config failure should be fixed by #9798). Spent some time messing around in Cloud Manager and didn't notice any UI differences or other changes in behavior.

**Review by jaalah-akamai — APPROVED** [review]

**Pull request merged**